tmk_core/protocol/vusb/vusb.c

```diff
@@ -509,7 +517,28 @@ const PROGMEM uchar shared_hid_report[] = {
     0x95, 0x01,                //   Report Count (1)
     0x75, 0x10,                //   Report Size (16)
     0x81, 0x00,                //   Input (Data, Array, Absolute)
-    0xC0                       // End Collection
+    0xC0,                      // End Collection
+#endif
+
+#ifdef PROGRAMMABLE_BUTTON_ENABLE
+    // Programmable buttons report descriptor
+    // padded to 16 bits to work with the existing machinery
+    0x05, 0x0C,                          // Usage Page (Consumer)
+    0x09, 0x01,                          // Usage (Consumer Control)
+    0xA1, 0x01,                          // Collection (Application)
+    0x85, REPORT_ID_PROGRAMMABLE_BUTTON, //   Report ID
+    0x09, 0x03,                          //   Usage (Programmable Buttons)
+    0xA1, 0x04,                          //   Collection (Named Array)
+    0x05, 0x09,                          //     Usage Page (Button)
+    0x19, 0x01,                          //     Usage Minimum (Button 1)
+    0x29, 0x05,                          //     Usage Maximum (Button 5)
+    0x15, 0x00,                          //     Logical Minimum
+    0x25, 0x01,                          //     Logical Maximum
+    0x95, 0x10,                          //     Report Count (16)
+    0x75, 0x01,                          //     Report Size (1)
+    0x81, 0x42,                          //     Input (Array, Variable, Absolute, Null State)
```

Comment:
If you are changing the implementation to send the button state as a bitmap, you need to change the descriptor accordingly — the item will no longer be an array, and the descriptor should look similar to what is used for NKRO or modifier keys in the 6KRO report. And the declared range of usages needs to have the same size as the report count.

A completely untested suggestion:
```suggestion
    0x19, 0x01,                          //     Usage Minimum (Button 1)
    0x29, 0x10,                          //     Usage Maximum (Button 16)
    0x15, 0x00,                          //     Logical Minimum
    0x25, 0x01,                          //     Logical Maximum
    0x95, 0x10,                          //     Report Count (16)
    0x75, 0x01,                          //     Report Size (1)
    0x81, 0x02,                          //     Input (Data, Variable, Absolute)
```